src/coreclr/jit/emitxarch.cpp

```diff
@@ -19860,6 +19860,7 @@ emitter::insExecutionCharacteristics emitter::getInsExecutionCharacteristics(ins
             break;
 
         case INS_pclmulqdq:
+        case INS_vpclmulqdq:
```

Comment:
nit: We shouldn't need a separate instruction here, the underlying encoding/opcode is the same.

We should never reach emit for `pclmulqdq` and `EA_32BYTE` or `EA_64BYTE` if the CPUID bit for `VPCLMULQDQ` isn't set

Reply:
The issue I ran into was that if I add the `Encoding_EVEX` flag to the existing instruction definition, we can emit EVEX encoding for 16-byte pclmulqdq if AVX-512 is supported but VPCLMULQDQ is not.  Having separate instructions and choosing between them in lowering solved that.  If you're saying it's possible/preferable to do the extra check in lsra, I can give that a shot.

Reply:
> If you're saying it's possible/preferable to do the extra check in lsra, I can give that a shot.

We already do this extra check for most instructions via `isEvexCompatibleHWIntrinsic`. You can see us query that here: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lsraxarch.cpp#L2143

and then if its false, we use `BuildEvexIncompatibleMask`: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lsraxarch.cpp#L2691-L2694

Updating `GenTree::isEvexCompatibleHWIntrinsic` (https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/gentree.cpp) is probably the right answer (which looks for `HW_Flag_NoEvexSemantics`).

Reply:
Gotcha.  It also required a check in `emitter::IsEvexEncodableInstruction`, but I got it working with a single instruction def.